src/hdmf/term_set.py

```diff
@@ -35,6 +37,7 @@ def __init__(self,
             msg = "Install linkml_runtime"
             raise ValueError(msg)
 
+        self.name = name
```

Comment:
@mavaylon1 what is `TermSet.name` being used for? I didn't see where the variable is actually being used in this PR. Is this preparing for some additional changes to come?

Reply:
I thought it could be useful for the user to name their termsets to more easily identify them without needing to open the schema.